src/Tizen.NUI/src/public/Layouting/LinearLayout.cs

```diff
@@ -495,70 +505,78 @@ private void MeasureVertical(MeasureSpecification widthMeasureSpec, MeasureSpeci
                     {
-                        childLayout.Owner.HeightSpecification = (int)(remainingHeight / childrenMatchParentCount);
                         needToMeasure = true;
                     }
-                }
 
-                if (needToMeasure == true)
-                {
-                    MeasureChildWithMargins(childLayout, widthMeasureSpec, new LayoutLength(0), heightMeasureSpec, new LayoutLength(0));
-                }
+                    if (remainingHeight > 0)
+                    {
+                        if ((childDesiredHeight == LayoutParamPolicies.MatchParent) || (useRemainingHeight))
+                        {
+                            // In MeasureChildWithMargins(), it is assumed that heightMeasureSpec includes Padding.Top and Padding.Bottom.
+                            // Therefore, Padding.Top and Padding.Bottom are added to heightMeasureSpec.Size before it is passed to MeasureChildWithMargins().
+                            heightMeasureSpec.SetSize(new LayoutLength((int)(remainingHeight / childrenMatchParentCount) + Padding.Top + Padding.Bottom));
+                            needToMeasure = true;
+                        }
+                    }
 
-                if (childWeight > 0)
-                {
-                    float childMeasuredHeight = childLayout.MeasuredHeight.Size.AsDecimal();
+                    if (needToMeasure == true)
+                    {
+                        MeasureChildWithMargins(childLayout, widthMeasureSpec, new LayoutLength(0), heightMeasureSpec, new LayoutLength(0));
+                    }
 
-                    if (childMeasuredHeight > 0)
+                    if (childWeight > 0)
                     {
-                        totalWeightLength += childMeasuredHeight;
+                        float childMeasuredHeight = childLayout.MeasuredHeight.Size.AsDecimal();
+
+                        if (childMeasuredHeight < 0)
+                        {
+                            totalWeightLength = Math.Max(totalWeightLength, totalWeightLength + childMargin.Top + childMargin.Bottom);
+                        }
+                        else
+                        {
+                            totalWeightLength = Math.Max(totalWeightLength, totalWeightLength + childMeasuredHeight + childMargin.Top + childMargin.Bottom);
+                        }
                     }
-                }
-            } // 2ND PHASE foreach
+                } // 2ND PHASE foreach
+            }
 
             // 3RD PHASE:
             //
             // We measure all weighted children whose owner has weight greater than 0.
             // After 3rd phase, all weighted children has height which is proportional to their weights
             // in remaining height of parent.
-            if (totalWeight > 0)
+            if ((isHeightExactly) && (totalWeight > 0))
```

Comment:
Could be better to print error about circular dependency when parent is wrapContent and child is MatchParent

Reply:
Thank you for your comment.
The error log has been added.